File: app/src/main/java/eu/toma/dev/playground/warcraft/di/WoWcomponent.java
package eu.toma.dev.playground.warcraft.di;

import javax.inject.Named;

import dagger.Component;
import eu.toma.dev.playground.GlobalComp;
import eu.toma.dev.playground.warcraft.api.MountService;

@Warcraft
@Component(modules = WoWModule.class, dependencies = GlobalComp.class)
public interface WoWcomponent
{

}

File: app/src/main/kotlin/warcraft/di/WoWcomponent.kt
package warcraft.di

import dagger.Component
import eu.toma.dev.playground.GlobalComp
import eu.toma.dev.playground.warcraft.api.MediaApi
import eu.toma.dev.playground.warcraft.api.WoWApi

@Warcraft
@Component(modules = arrayOf(WoWModule::class), dependencies = arrayOf(GlobalComp::class))
interface WoWcomponent {
    fun provideApi(): WoWApi

    fun mediaApi(): MediaApi
}
